### ISSUE TITLE
Test adapter not found for >= .net5.0 test containers

### ISSUE THREAD

**Issue by fkpama** [microsoft/vstest#4842]
## Description

Hello,

I'm working on a [test adapter for benchmarks](https://github.com/fkpama/Benchmarker/tree/dev) that I'm trying to execute using the vstest.console command line with /TestAdapterPath.

The test adapter is discovered for .net461 unit test projects and the tests are correctly ran (don't mind the exception in the logs, that one is in the adapter, on my end)
![image](https://github.com/microsoft/vstest/assets/37318755/f21326f0-3879-4324-98aa-ddbcc140987e)

The test adapter is *not* discovered for >= .net5.0 test projects and the user receives a warning that no tests are available
![image](https://github.com/microsoft/vstest/assets/37318755/f10b4adf-8cd7-485a-9a08-4fe8b25efbdc)

## Steps to reproduce

1) Create a TestAdapter project (TFM: netstandard2.0)
2) Create an MsTest project for that adapter (TFM: net461, any >= net5.0)
3) Build the projects
4) open a regular console commands (outside vs env but should be the same)
5) launch vstest.console from devenv directory with --TestAdapterPath:<path to test adapter dll> and the > net461 < unit test project output ====> OK, the test adapter is correctly loaded
6) launch vstest.console from devenv directory with --TestAdapterPath:<path to test adapter dll> and the > net5.0 < unit test project output

## Expected behavior

The test adapter is discovered and the RunTests method is called

## Actual behavior

The test adapter is *not* discovered and the RunTests method is *not* called

## Diagnostic logs

[log-net7.0-KO.log](https://github.com/microsoft/vstest/files/14008992/log-net7.0-KO.log)
[log-net461-OK.txt](https://github.com/microsoft/vstest/files/14008993/log-net461-OK.txt)

## Environment
![image](https://github.com/microsoft/vstest/assets/37318755/7a0c69b4-dffd-41ea-b283-748d568fc99a)

**Comment by nohwnd** [comment]
I am surprised this works for .net framework with the full path to the adapter, because it should be a directory of where the adapter is placed (not sure why we did not change that), other than that I don't see a reason why it should not work, no errors specific to your adapter in the KO log, it just never tries to load that adapter dll.

**Comment by nohwnd** [comment]
Please try providing the adapters path as directory only, not the dll, and let me know if that helped. :)

**Comment by fkpama** [comment]
Hi @nohwnd, thanks for the quick help.

Yes I tried both without success.

From my understanding, the behavior you describe is indeed [the default](https://github.com/microsoft/vstest/blob/f4015340ec4139197f3f0516c00d0382dec0177b/src/Microsoft.TestPlatform.Client/TestPlatform.cs#L323) behavior. However, I think the framework is supposed to [handle both use cases when the TestAdapterLoadingStrategy is set to Explicit](https://github.com/microsoft/vstest/blob/f4015340ec4139197f3f0516c00d0382dec0177b/src/Microsoft.TestPlatform.Client/TestPlatform.cs#L334).

From my little investigation, the problem seems to come from the fact that the test runtime provider does not send back the path of the test adapter when [sending the TestExecution.Initialize command to the testhost](https://github.com/microsoft/vstest/blob/f4015340ec4139197f3f0516c00d0382dec0177b/src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs#L448), but I can't even compile the project so I figured it would be quicker to raise an issue

Thanks for the help though, if you have anything else you want me to try just let me know

**Comment by nohwnd** [comment]
That was other thing that I was suspecting, but it should send the info eventually. 

Easiest way to debug this kind of issue is via the Playground project we have in this repo, you will set values in the EnvironmentVariables.cs and run that project and it should build everything for you, and automatically attach the sub processes to your Visual Studio (if you are using VS for your development). 

What problem do you have building this repo? It is because you are using VS 2019 and we require 2022?

**Comment by fkpama** [comment]
@nohwnd, well, Before I gave up I had to:

1) switch to main branch because the master [requires csharp 9](https://github.com/microsoft/vstest/blob/f4015340ec4139197f3f0516c00d0382dec0177b/global.json#L3)
2) [find and disable the analyzers](https://github.com/microsoft/vstest/blob/f4015340ec4139197f3f0516c00d0382dec0177b/Directory.Build.props#L41C5-L46C43) that requires .net 4.8 (I'm on 4.7)
3) remove the x86/arm dependencies of the Playground project because of a dnxcore50/netcore50 requirement (similar to 4)
4) I switched back to v17.8.0 and had this error,

![image](https://github.com/microsoft/vstest/assets/37318755/274f1c51-5b6b-4fbf-87b2-210f12e02e6a)

Long story short, the rainbow in my console window didn't want to shrink after each each "fix" so I said "not today" :)

I don't doubt it can be done if you really want to, but let's just say it's not the simplest project out there to clone for a quickfix :)

**Comment by fkpama** [comment]
@nohwnd, thanks a lot for that.

Yes it's weird that it went unnoticed for so long indeed. Ok I understand the concern so I will do the necessary modifications on my end then.

This ticket can be closed

Thanks again for the quick help
